### PR TITLE
Refactor: CMake Integration tests

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lfortran/lfortran/issues/2152

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
The old `4` commits are part of the PR https://github.com/lfortran/lfortran/pull/2151. I will rebase this PR after merging/completion of https://github.com/lfortran/lfortran/pull/2151.

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -928,7 +924,8 @@ RUN(NAME array_01 LABELS gfortran llvm)
 RUN(NAME array_02 LABELS gfortran llvm wasm)
 
 RUN(NAME do_loop_02 LABELS gfortran llvm)
-RUN(NAME do_loop_03 LABELS gfortran llvmUseLoopVariableAfterLoop)
+RUN(NAME do_loop_03 LABELS gfortran llvm
+    EXTRA_ARGS --use-loop-variable-after-loop)
```
Comment (Pranavchiku) [inline integration_tests/CMakeLists.txt:928]: Why do we need this, I feel the `llvmUseLoopVariableAfterLoop` works,no?

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -1020,16 +1000,21 @@ RUN(NAME character_01 LABELS gfortran llvm)
...
 RUN(NAME c_ptr_01 LABELS gfortran llvm NOFAST)
 
-RUN(NAME implicit_argument_casting_01 LABELS llvmImplicitArgument)
-
 RUN(NAME arrayitem_01 LABELS gfortran llvm)
 
-RUN(NAME implicit_argument_casting_01 LABELS gfortran IMPLICIT_ARGS)
+RUN(NAME implicit_argument_casting_01 LABELS gfortran llvm
+    EXTRA_ARGS --implicit-argument-casting
+    GFORTRAN_ARGS -fallow-argument-mismatch)
```
Comment (Pranavchiku) [inline integration_tests/CMakeLists.txt:1007]: same comment here.

File: integration_tests/CMakeLists.txt [hunk 3]
```diff
@@ -1020,16 +1000,21 @@ RUN(NAME character_01 LABELS gfortran llvm)
...
+    EXTRA_ARGS --use-loop-variable-after-loop --implicit-typing)
 
-RUN(NAME cpp_pre_01 LABELS gfortran llvm c CPP_PRE)
-RUN(NAME cpp_pre_02 LABELS gfortran llvm c wasm CPP_PRE)
+RUN(NAME cpp_pre_01 LABELS gfortran llvm c
+    EXTRA_ARGS --cpp
+    GFORTRAN_ARGS -cpp)
+RUN(NAME cpp_pre_02 LABELS gfortran llvm c wasm
+    EXTRA_ARGS --cpp
+    GFORTRAN_ARGS -cpp)
```
Comment (Pranavchiku) [inline integration_tests/CMakeLists.txt:1017]: and here.

File: integration_tests/CMakeLists.txt (hunk 1 above)
Comment (certik) [reply integration_tests/CMakeLists.txt:928]: I think this new approach is more robust -- we can add any options we need without changing the cmake macros, which is always a fragile operation, as we need to manually test that everything works. This new approach should be equivalent --- but we need to carefully test it to make sure it actually works.

**Review by certik — APPROVED** [review]
I think that this is fine, assuming it works. Very careful testing is needed to make sure the CI and these tests didn't break.

File: integration_tests/CMakeLists.txt (hunk 1 above)
Comment (Pranavchiku) [reply integration_tests/CMakeLists.txt:928]: okay, yes. This seems to be a better option than manually adding a new flag everytime which can be fragile as well.

**Pull request merged**